src/store/peer/sagas.ts

```diff
@@ -62,10 +74,92 @@ export function* deletePeer(action: ReturnType<typeof actions.deletedPeer.reques
   }
 }
 
+export function* saveGroups(action: ReturnType<typeof actions.saveGroups.request>): Generator {
+  try {
+    yield put(actions.setSavedGroups({
+      loading: true,
+      success: false,
+      failure: false,
+      error: null,
+      data: null
+    }))
+
+    const currentGroups = [...(yield select(state => state.group.data)) as Group[]]
+
+    const peerGroupsToSave = action.payload.payload
+
+    let groupsToSave = [] as Group[]
+    let groupsNoId = [] as Group[]
+
+    groupsToSave = groupsToSave.concat(
+        currentGroups
+            .filter(g => peerGroupsToSave.groupsToRemove.includes(g.ID || ''))
+            .map(g => ({
+              ID: g.ID,
+              Name: g.Name,
+              Peers: (g.Peers as GroupPeer[]).filter(p => p.ID !== peerGroupsToSave.ID).map(p => p.ID) as string[]
+            }))
+    )
+
+    groupsToSave = groupsToSave.concat(
+        currentGroups
+            .filter(g => peerGroupsToSave.groupsToAdd.includes(g.ID || ''))
+            .map(g => ({
+              ID: g.ID,
+              Name: g.Name,
+              Peers: g.Peers ? [...(g.Peers as GroupPeer[]).map((p:GroupPeer) => p.ID), peerGroupsToSave.ID] : [peerGroupsToSave.ID]
+            }))
+    )
+
+    groupsNoId = peerGroupsToSave.groupsNoId.map(g => ({
+      Name: g,
+      Peers: [peerGroupsToSave.ID]
+    }))
+
+    if (!groupsNoId.length && !groupsToSave.length) {
+      return
+    }
+
+    const responsesGroup = yield all(groupsToSave.map(g => call(serviceGroup.editGroup, {
```

Comment:
Codacy has a fix for the issue: [Replace `groupsToSave.map(g·=>` with `⏎······groupsToSave.map((g)·=>⏎·······`](https://app.codacy.com/gh/netbirdio/dashboard/pullRequest?prid=9560373)
```suggestion
    const responsesGroup = yield all(
```